.circleci/config.yml

```diff
@@ -0,0 +1,98 @@
+version: 2
+jobs:
+  pch_and_sql:
+    docker:
+      - image: trinitycore/circle-ci:master-buildpacks-disco
+      - image: circleci/mysql:5.7
+        environment:
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+          MYSQL_ROOT_PASSWORD: ''
+          MYSQL_DATABASE: test_mysql
+    steps:
+      - run:
+          name: Checkout
+          command: |
+            /scripts/checkout.sh
+      - run:
+          name: Setup
+          command: |
+            mkdir bin
+            cd bin
+            cmake ../ -DWITH_WARNINGS=1 -DWITH_COREDEBUG=0 -DUSE_COREPCH=1 -DUSE_SCRIPTPCH=1 -DTOOLS=1 -DSCRIPTS=dynamic -DSERVERS=1 -DNOJEM=0 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS="-Werror" -DCMAKE_CXX_FLAGS="-Werror" -DCMAKE_C_FLAGS_DEBUG="-DNDEBUG" -DCMAKE_CXX_FLAGS_DEBUG="-DNDEBUG" -DCMAKE_INSTALL_PREFIX=check_install
+            cd ..
+      - run:
+          name: SQL checks
+          command: |
+            chmod +x contrib/check_updates.sh
+            dockerize -wait tcp://127.0.0.1:3306 -timeout 120s
+            mysql -h 127.0.0.1 -uroot < sql/create/create_mysql.sql
+            mysql -h 127.0.0.1 -uroot auth < sql/base/auth_database.sql
+            ./contrib/check_updates.sh auth master auth 127.0.0.1
+            mysql -h 127.0.0.1 -uroot characters < sql/base/characters_database.sql
+            ./contrib/check_updates.sh characters master characters 127.0.0.1
+            mysql -h 127.0.0.1 -uroot world < sql/base/dev/world_database.sql
+            mysql -h 127.0.0.1 -uroot hotfixes < sql/base/dev/hotfixes_database.sql
+            cat sql/updates/world/master/*.sql | mysql -h 127.0.0.1 -uroot world
+            cat sql/updates/hotfixes/master/*.sql | mysql -h 127.0.0.1 -uroot hotfixes
+            mysql -h 127.0.0.1 -uroot < sql/create/drop_mysql.sql
+      - run:
+          name: Build
+          command: |
+            cd bin
+            make -j 4 -k && make install
+            cd check_install/bin
+            ./bnetserver --version
+            ./worldserver --version
+  nopch:
+    docker:
+      - image: trinitycore/circle-ci:master-buildpacks-disco
+    steps:
+      - run:
+          name: Requirements
+          command: |
+            sudo mkdir -p /root/.ccache
+            sudo chown -Rv circleci:circleci /root
+      - run:
+          name: Checkout
+          command: |
+            /scripts/checkout.sh
+      - restore_cache:
+          keys:
+            - nopch-{{ .Branch }}-{{ .Revision }}
+            - nopch-{{ .Branch }}
+            - nopch
```

Comment:
master should have a different key (we might change that for 3.3.5 branch too )

Reply:
i would drop the nopch key and just use 
```
            - nopch-{{ .Branch }}-{{ .Revision }}
            - nopch-{{ .Branch }}
```